lib/matplotlib/backends/backend_ps.py

```diff
@@ -867,18 +867,24 @@ def _print_figure(
         # find the appropriate papertype
         width, height = self.figure.get_size_inches()
         if papertype == 'auto':
-            papertype = _get_papertype(
-                *orientation.swap_if_landscape((width, height)))
-        paper_width, paper_height = orientation.swap_if_landscape(
-            papersize[papertype])
+            _api.warn_deprecated("3.8", name="papertype='auto'",
+                                 addendum="Pass an explicit paper type, or omit the "
+                                 "*papertype* argument entirely.")
+            papertype = _get_papertype(*orientation.swap_if_landscape((width, height)))
 
-        if mpl.rcParams['ps.usedistiller']:
-            # distillers improperly clip eps files if pagesize is too small
-            if width > paper_width or height > paper_height:
-                papertype = _get_papertype(
-                    *orientation.swap_if_landscape((width, height)))
-                paper_width, paper_height = orientation.swap_if_landscape(
-                    papersize[papertype])
+        if is_eps:
+            paper_width, paper_height = width, height
+        else:
+            paper_width, paper_height = orientation.swap_if_landscape(
+                papersize[papertype])
+
+            if mpl.rcParams['ps.usedistiller']:
+                # distillers improperly clip eps files if pagesize is too small
```

Comment:
This comment seems incongruous with this being in the `else` block of `if is_eps:`

If this code is specifically targeted to fix eps, then it would have to be in the if block, and if it is just wrong, perhaps it could be taken out entirely.

It also relies on the `_get_papertype` function which the point of removing `auto` is to remove that function because it is problematic in implementation.

Reply:
Ah good point, since as the first commit notes, the distiller ignores the paper type and asks the distiller to crop, this is probably something that could be dropped. But I guess I will have to test out a few combinations to be sure.